examples/unreliable-datagram/src/bin/datagram_receiver.rs

```diff
@@ -0,0 +1,61 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use core::task::Poll;
+use s2n_quic::{
+    client::Connect,
+    provider::datagram::{default::Endpoint, default::Receiver},
+    Client,
+};
+use std::{error::Error, net::SocketAddr};
+
+/// NOTE: this certificate/key pair is to be used for demonstration purposes only!
+pub static CERT_PEM: &str = include_str!(concat!(
+    env!("CARGO_MANIFEST_DIR"),
+    "/../../quic/s2n-quic-core/certs/cert.pem"
+));
+pub static KEY_PEM: &str = include_str!(concat!(
+    env!("CARGO_MANIFEST_DIR"),
+    "/../../quic/s2n-quic-core/certs/key.pem"
+));
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn Error>> {
+    // Create a datagram provider that has recv queue capacity
+    let datagram_provider = Endpoint::builder()
+        .with_recv_capacity(200)?
+        .build()
+        .unwrap();
+
+    // Build an `s2n_quic::Client`
+    let client = Client::builder()
+        .with_tls((CERT_PEM, KEY_PEM))?
+        .with_io("0.0.0.0:0")?
+        .with_datagram(datagram_provider)?
+        .start()?;
+
+    let addr: SocketAddr = "127.0.0.1:4433".parse()?;
+    let connect = Connect::new(addr).with_server_name("localhost");
+    let mut connection = client.connect(connect).await?;
+
+    // The poll_fn function expects a Poll enum wrapped around a Result. Here we call
+    // the desired recv method in a closure and modify the returned value
+    // to be what the poll_fn expects.
```

Comment:
This is just describing what Rust code is doing and not what s2n-quic specific behavior is. That doesnt seem very useful for someone looking for a s2n-quic example.